sdv/data_processing/data_processor.py

```diff
@@ -465,6 +465,15 @@ def _fit_numerical_formatters(self, data):
                 )
                 self.formatters[column_name].learn_format(data[column_name])
 
+    def _fit_datetime_formatters(self, data):
```

Comment:
This method is so similar to `_fit_numerical_formatters` that we could probably combine them into a `_fit_formatters` and just have different if statements and one loop

Reply:
I was doubting between that option too. Will address it asap!